components/join-us-modal/join-us-modal.jsx

```diff
@@ -0,0 +1,326 @@
+import React from "react";
+import { Link } from "react-router-dom";
+import Modal from "react-modal";
+import { Form } from "react-formbuilder";
+import qs from "qs";
+import Step from "./step.jsx";
+import user from "../../js/app-user";
+import utility from "../../js/utility";
+import Service from "../../js/service";
+import createFields from "./form/create-fields";
+
+Modal.setAppElement("#app");
+
+const SIGN_UP_PARAM = `sign_up`;
+const EDIT_NEW_PROFILE_PARAM = `edit_new_profile`;
+
+class JoinUsModal extends React.Component {
+  constructor(props) {
+    super(props);
+    this.totalStep = 3;
+    this.state = {
+      user: {},
+      formValues: {},
+      currentStep: 1,
+      showModal: false,
+      submitting: false,
+      showConfirmation: false
+    };
+  }
+
+  openModal() {
+    this.setState({ showModal: true }, () => {
+      // a hack to disable 'email' field
+      // because react-formbuilder doesn't allow setting a field as disabled
```

Comment:
hm, what are we using `disabled` for, in this case? (we can add prop passthrough to the formbuilder, but if we can achieve the same thing with a class, that's probably better?)

Reply:
We want to display user's email in a non-editable field. ^ was the easiest solution I could come up with without touching formbuilder.

> we can add prop passthrough to the formbuilder, but if we can achieve the same thing with a class, that's probably better?

Can you elaborate on what you meant by using a class?

Reply:
a CSS class, but I guess that wouldn't prevent the field from being editable so... no dice on that!